hybrid_planning_demo/src/global_mtc_planner.cpp

```diff
@@ -0,0 +1,132 @@
+  planning_solution.group_name = "ur_manipulator";
+
+  task_ = std::make_shared<moveit::task_constructor::Task>();
+  Task& t = *task_;
+  t.stages()->setName("global_mtc_task");
+  t.loadRobotModel(shared_from_this());
+
+  // Sampling planner
+  auto sampling_planner = std::make_shared<solvers::PipelinePlanner>(shared_from_this());
+  sampling_planner->setProperty("goal_joint_tolerance", 1e-5);
+
+  // Cartesian planner
+  auto cartesian_planner = std::make_shared<solvers::CartesianPath>();
+  cartesian_planner->setMaxVelocityScaling(1.0);
+  cartesian_planner->setMaxAccelerationScaling(1.0);
+  cartesian_planner->setStepSize(.01);
+
+  // Set task properties
+  t.setProperty("group", "ur_manipulator");
+  // t.setProperty("eef", eef_name_);
+  // t.setProperty("hand", hand_group_name_);
+  // t.setProperty("hand_grasping_frame", hand_frame_);
+  t.setProperty("ik_frame", "tcp_welding_gun_link");
+
+  /****************************************************
+   *                                                  *
+   *               Current State                      *
+   *                                                  *
+   ***************************************************/
+  using namespace moveit::task_constructor::stages;
+  {
+    auto current_state = std::make_unique<stages::CurrentState>("current state");
+    t.add(std::move(current_state));
+  }
+
+  // Move to joint state constraints specified in goal request
+  // NOTE: this assumes only joint constraints are specified
+  {
+    auto stage = std::make_unique<stages::MoveTo>("move to goal", sampling_planner);
+    stage->setGroup("ur_manipulator");
+    std::map<std::string, double> goal_state;
+    for (const auto& jc : planning_problem.goal_constraints[0].joint_constraints)
+    {
+      goal_state[jc.joint_name] = jc.position;
+    }
+    stage->setGoal(goal_state);
+    t.add(std::move(stage));
+  }
+
+  /******************************************************
+   *          Relative Motion                           *
+   *****************************************************/
+  // {
+  //   auto stage = std::make_unique<stages::MoveRelative>("relative motion", cartesian_planner);
+  //   stage->properties().configureInitFrom(Stage::PARENT, { "group" });
+  //   stage->setMinMaxDistance(.01, .05);
+  //   stage->setIKFrame("tcp_welding_gun_link");
+  //   stage->properties().set("marker_ns", "retreat");
+  //   geometry_msgs::msg::Vector3Stamped vec;
+  //   vec.header.frame_id = "world";
+  //   vec.vector.z = -1.0;
+  //   stage->setDirection(vec);
+  //   t.add(std::move(stage));
+  // }
+
+  constexpr size_t max_solutions = 1;
+  t.plan(max_solutions);
+  if (t.numSolutions() == max_solutions)
+  {
+    moveit_task_constructor_msgs::msg::Solution solution;
+    t.solutions().front()->fillMessage(solution, &t.introspection());
+    auto& solution_traj = planning_solution.trajectory;
+    for (const auto& sub_traj : solution.sub_trajectory)
```

Comment:
this is just a hacky way to fill in the solution trajectory, we should implement something alike the parsing function in the ExecuteTaskSolutionCapability to process the solution